### PR TITLE
Gra 1217 tests

### PR DESCRIPTION
**Pull request by mattkasun**
## Describe your changes
fixed unit tests

## Provide testing steps
see GRA-1217

## Checklist before requesting a review
- [x] My changes affect only 10 files or less.
- [x] I have performed a self-review of my code and tested it.
- [ ] If it is a new feature, I have added thorough tests, my code is <= 1450 lines.
- [x] If it is a bugfix, my code is <= 200 lines.
- [ ] My functions are <= 80 lines.
- [ ] I have had my code reviewed by a peer.
- [ ] My unit tests pass locally.
- [ ] Netmaker is awesome.

### CODE REVIEW

File: functions/helpers_test.go [hunk 1]
```diff
@@ -39,40 +41,29 @@ func TestMain(m *testing.M) {
...
+	assert.Nil(t, err)
 	exists, err = logic.NetworkExists(testNetwork.NetID)
-	if err != nil {
-		t.Fatalf("expected nil, received err: %s", err)
-	}
-	if !exists {
-		t.Fatalf("expected network to exist in database")
-	}
+	assert.Nil(t, err)
+	assert.NotNil(t, exists)
```
Comment (abhishek9686) [inline functions/helpers_test.go:58]: shouldn't this be `assert.True`

**Review by abhishek9686 — CHANGES_REQUESTED** [review]

**Comment by 0xdcarns** [comment]
missing description and testing steps

File: controllers/network_test.go [hunk 2]
```diff
@@ -40,6 +40,8 @@ func TestMain(m *testing.M) {
 			logger.Log(3, "received node update", update.Action)
 		}
 	}()
+	os.Exit(m.Run())
```
Comment (0xdcarns) [inline controllers/network_test.go:43]: network tests should not rely on nodes, there is no point in this main

**Review by 0xdcarns — CHANGES_REQUESTED** [review]

**Review by 0xdcarns — CHANGES_REQUESTED** [review]

File: controllers/network_test.go (hunk 2 above)
Comment (mattkasun) [reply controllers/network_test.go:43]: a TestMain is required in every package to prevent tests from deadlocking

**Review by theguy951357 — APPROVED** [review]
unit tests run and pass

File: controllers/network_test.go (hunk 2 above)
Comment (mattkasun) [reply controllers/network_test.go:43]:
```
mkasun@endeavour ~/netmaker1 (GRA-1217-tests)> grep -R TestMain *
controllers/network_test.go:func TestMain(m *testing.M) {
functions/helpers_test.go:func TestMain(m *testing.M) {
logic/pro/networkuser_test.go:func TestMain(m *testing.M) {
logic/host_test.go:func TestMain(m *testing.M) {
test/api_test.go.bak:func TestMain(m *testing.M) {
```
```
mkasun@endeavour ~/netmaker1 (GRA-1217-tests)> grep -R m.Run  *
controllers/network_test.go:    os.Exit(m.Run())
functions/helpers_test.go:      os.Exit(m.Run())
logic/pro/networkuser_test.go:  os.Exit(m.Run())
logic/host_test.go:     os.Exit(m.Run())
grep: netclient/ncutils/windowsdaemon/winsw.exe: binary file matches
test/api_test.go.bak:   os.Exit(m.Run())
mkasun@endeavour ~/netmaker1 (GRA-1217-tests)> 

```

**Comment by mattkasun** [comment]
GRA-1350 covers failed test

**Review by afeiszli — APPROVED** [review]

**Review by theguy951357 — APPROVED** [review]
tests pass
critical checklist 100%

**Pull request merged**